planning/behavior_path_planner/src/behavior_path_planner_node.cpp

```diff
@@ -586,7 +586,29 @@ void BehaviorPathPlannerNode::run()
   const auto output = bt_manager_->run(planner_data);
 
   // path handling
-  const auto path = getPath(output, planner_data);
+  auto path = getPath(output, planner_data);
+
+  // check if path orientation is valid
+  if (!path->points.empty()) {
+    const auto closest_path_point_idx =
+      motion_utils::findNearestIndex(path->points, planner_data->self_pose->pose.position);
+    const auto yaw_dev = tier4_autoware_utils::calcYawDeviation(
+      path->points.at(closest_path_point_idx).point.pose, planner_data->self_pose->pose);
+
+    if (std::abs(yaw_dev) > M_PI / 2.0) {
+      RCLCPP_WARN(
+        get_logger(), "Path orientation is invalid. Path yaw deviation: %f (deg)",
+        yaw_dev * 180 / M_PI);
+      if (!planner_data_->prev_output_path->points.empty()) {
+        path = planner_data_->prev_output_path;
+      } else {
+        return;
+      }
+    } else {
```

Comment:
Sorry to convert the PR to draft.

Using previously planned path resolves the issue, but ideally using previous path should be avoided since the current planning with the current dynamic situation is not applied to the planning result.

This issue is known one. During lane_following in behavior_path_planner, the nearest lanelet to ego is searched and the path will start from the lanelet. Therefore, if some functions like drivable area expansion moves the ego pose CoG to neighbour lanelets, the path will suddenly start from the neighbour lanelet.

Rather than hiding an issue with previous path, we should fix this known issue fundamentally.